src/librustc_const_math/float.rs

```diff
@@ -203,3 +203,8 @@ impl ::std::ops::Neg for ConstFloat {
         ConstFloat { bits, ty: self.ty }
     }
 }
+
+/// This is `(f32::MAX + 0.5 ULP)` as an integer. Numbers greater or equal to this
+/// are rounded to infinity when converted to `f32`.
+pub const MAX_F32_PLUS_HALF_ULP: u128 =
+  (1 << Single::PRECISION + 1) - 1 << (Single::MAX_EXP - Single::PRECISION as i16);;
```

Comment:
There's an extra `;` on this line. Also, to repeat my question, what's the `+ 1` for? The previous constant looked like a subtraction of two powers of 2. Maybe there are "not enough parens" both around shifts and around nontrivial shift operands?

Reply:
I don't think of this constant as a difference of powers of two, but rather as PRECISION + 1 bits set (maximum significand + 1 bit for the half ulp), multiplied by maximum exponent (accounting for normalization).

Reply:
Oh then I guess your choice of parens isn't uniform and that's what confused me.

Reply:
I added a bunch of parens and an explanatory comment now (and also removed the duplicate `;`).

Reply:
I added a bunch of parens and an explanatory comment.